source/states/PlayerAirState.hx

```diff
@@ -0,0 +1,103 @@
+package states;
+
+import flixel.FlxG;
+import flixel.FlxObject;
+import flixel.input.keyboard.FlxKey;
+
+class PlayerAirState extends BaseState
+{
+  static private var JUMP:Int = 400;
+
+  public function new()
+  {
+    super();
+  }
+
+  private override function transition(object:FlxObject):Bool
+  {
+    var player = cast(object, Player);
+
+    if (player.isOnGround())
+    {
+      nextState = new PlayerGroundState();
+      return true;
+    }
+
+    if (player.velocity.y >= 0)
+    {
+      nextState = new PlayerFallState();
+      return true;
+    }
+
+    return super.transition(object);
+  }
+
+  public override function enter(object:FlxObject):Void
+  {
+      var player = cast(object, Player);
+      player.velocity.y = -JUMP;
+  }
```

Comment:
I don't think that the PlayerAirState's default behavior should be to make the character jump. Any action that makes the player lose contact with the ground will put them in the air state, which goes beyond just jumping.